### PR TITLE
refactor filter by rows

### PR DESCRIPTION
**Pull request by hAz5**
refactor filter by rows

### CODE REVIEW

File: src/FakeQueryBuilder.php [hunk 1]
```diff
@@ -98,15 +98,15 @@ public function whereNotNull($columns, $boolean = 'and')
         return $this;
     }
 
-    public function whereBetween($column, array $values, $boolean = 'and', $not = false)
+    public function whereBetween($column, iterable $values, $boolean = 'and', $not = false)
```
Comment (hAz5) [inline src/FakeQueryBuilder.php:100]: this change for compatible with the parent Builder  class I got exception in running

File: src/FakeQueryBuilder.php [hunk 2]
```diff
@@ -98,15 +98,15 @@ public function whereNotNull($columns, $boolean = 'and')
...
     {
         $this->recordedWhereBetween[] = [$column, $values];
 
         return $this;
 
     }
 
-    public function whereNotBetween($column, array $values, $boolean = 'and')
+    public function whereNotBetween($column, iterable $values, $boolean = 'and')
```
Comment (hAz5) [inline src/FakeQueryBuilder.php:107]: this change for compatible with the parent Builder  class I got exception in running

**Review by milwad-dev — APPROVED** [review]
merge

**Comment by imanghafoori1** [comment]
tests do not pass

**Comment by hAz5** [comment]
> tests do not pass

I changed patterns and I think it's better than its old pull request
please don't be hesitate if its need more changes

**Comment by imanghafoori1** [comment]
are you finished?

**Pull request merged**